samples/simple_txrx/src/main.c

```diff
@@ -0,0 +1,290 @@
+/*
+ * Copyright (c) 2024 Silicon Laboratories Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/shell/shell.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+
+#include "rail.h"
+#include "rail_config.h"
+#include "pa_conversions_efr32.h"
+
+LOG_MODULE_REGISTER(app);
+
+#ifdef RAIL0_CHANNEL_GROUP_1_PROFILE_WISUN_OFDM
+#  if !defined(HARDWARE_BOARD_HAS_EFF)
+     BUILD_ASSERT(SL_RAIL_UTIL_PA_SELECTION_SUBGHZ == RAIL_TX_POWER_MODE_OFDM_PA,
+		  "Please use the OFDM PA settings in the sl_rail_util_pa_config.h for OFDM phys");
+#  endif
+#  if defined(HARDWARE_BOARD_HAS_EFF) && RAIL_SUPPORTS_EFF
+     BUILD_ASSERT(SL_RAIL_UTIL_PA_SELECTION_SUBGHZ >= RAIL_TX_POWER_MODE_OFDM_PA_EFF_30DBM,
+		  "Please use the OFDM PA for EFF settings in the sl_rail_util_pa_config.h for OFDM phys");
+#  endif
+#endif
+
+static const struct gpio_dt_spec sw0 = GPIO_DT_SPEC_GET(DT_ALIAS(sw0), gpios);
+static const struct gpio_dt_spec led_rx = GPIO_DT_SPEC_GET(DT_ALIAS(led0), gpios);
+#if DT_NODE_EXISTS(DT_ALIAS(led1))
+static const struct gpio_dt_spec led_tx = GPIO_DT_SPEC_GET(DT_ALIAS(led1), gpios);
+#else
+static const struct gpio_dt_spec led_tx = led_rx;
+#endif
+
+enum {
+	EV_RAIL_RX     = BIT(0),
+	EV_BTN_PRESSED = BIT(1),
+};
+
+struct {
+	RAIL_Handle_t rail_handle;
+	struct k_event rx_event;
+	int channel;
+	const uint8_t *payload;
+	int payload_len;
+} app_ctx;
+
+void rx_packets(RAIL_Handle_t rail_handle)
+{
+	uint8_t rx_frame[32];
+	RAIL_RxPacketHandle_t handle;
+	RAIL_RxPacketInfo_t info;
+	RAIL_Status_t status;
+
+	for (;;) {
+		handle = RAIL_GetRxPacketInfo(rail_handle, RAIL_RX_PACKET_HANDLE_OLDEST_COMPLETE,
+					      &info);
+		if (handle == RAIL_RX_PACKET_HANDLE_INVALID) {
+			return;
+		}
+		if (info.packetBytes < sizeof(rx_frame)) {
+			RAIL_CopyRxPacket(rx_frame, &info);
+		}
+		status = RAIL_ReleaseRxPacket(rail_handle, handle);
+		if (status) {
+			LOG_ERR("RAIL_ReleaseRxPacket(): %d", status);
+		}
+		if (info.packetBytes < sizeof(rx_frame)) {
+			LOG_HEXDUMP_INF(rx_frame, info.packetBytes, "rx data:");
+		} else {
+			LOG_INF("rx: skip large packet");
+		}
+		gpio_pin_toggle_dt(&led_rx);
+	}
+}
+
+void tx_packet(RAIL_Handle_t rail_handle, int channel, const uint8_t *payload, int len)
+{
+	RAIL_Status_t status;
+	int ret;
+
+	ret = RAIL_WriteTxFifo(rail_handle, payload, len, true);
+	if (ret != len) {
+		LOG_ERR("RAIL_WriteTxFifo(): %d", ret);
+		return;
+	}
+	gpio_pin_toggle_dt(&led_tx);
```

Comment:
Yeah, I am not super happy with this behavior. However the original one (toggle the led each time a frame is sent) was also uncommon.

Ideally, we would like to behave like the led of an ethernet port. However, I don't want to clutter this sample code with features not related to the radio.

What is your opinion?

Reply:
I understand your point and we can say that we go that way. I only asked this because I don't know how much likeness we would like to keep with the original SiSDK one.